src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -531,25 +533,27 @@ parseIdentifierAndBinder =
 -- | Parse a binder
 parseBinder :: TokenParser Binder
 parseBinder =
+  flattenPositionedBinder <$>
   withSourceSpan
     PositionedBinder
     ( P.buildExpressionParser operators
     . buildPostfixParser postfixTable
     $ parseBinderAtom
     )
   where
-  operators =
-    [ [ P.Infix (P.try (indented *> parseOpBinder P.<?> "binder operator") >>= \op ->
-          return (BinaryNoParensBinder op)) P.AssocRight
+    flattenPositionedBinder (PositionedBinder p c (PositionedBinder p' _ b)) | p == p' = PositionedBinder p c b
```

Comment:
The PR seems good to me, but I'd really like to be able to remove this. Seems the parser should be able to guarantee we only add one `PositionedBinder`.